qmp/socket.go

```diff
@@ -66,9 +68,11 @@ func NewSocketMonitor(network, addr string, timeout time.Duration) (*SocketMonit
 		return nil, err
 	}
 
+	listeners := make(chan struct{})
+	close(listeners)
 	mon := &SocketMonitor{
-		c:         c,
-		listeners: new(int32),
+		c:           c,
+		noListeners: listeners,
```

Comment:
The naming here is inherently paradoxical. noListeners = listeners is just going to confuse everyone. I like the convention of affixing Ch or Chan to the names of channels, so consider that too. But see my next comment first.